### ISSUE TITLE
[BUG] useLocalStorage get Unhandled Runtime Error in nextjs

### ISSUE THREAD

**Issue by shaddollxz** [juliencrn/usehooks-ts#631]
### Describe the bug

i use useLocalStorage in my project, while i get the default localStorage value(is null) in SSR, but get it in client with some value, i got a error `Text content does not match server-rendered HTML.` how to fix this bug

### To Reproduce

there is my code example: 
https://codesandbox.io/p/github/shaddollxz/use-local-storage/draft/magical-jackson?file=%2Fapp%2Fprovider.tsx%3A30%2C1&workspaceId=af9ba02f-cf52-4c89-8862-6db030644648

### Expected behavior

not show this error

**Comment by vimalsonara** [comment]
Can't access codesandbox and maybe it's hydration error check rendering.

**Comment by shaddollxz** [comment]
> Can't access codesandbox and maybe it's hydration error check rendering.

yes, it is hydration error,and sandbox link now updated; is there any way to resolve it? i am new to nextjs.

**Comment by vimalsonara** [comment]
You can resolve this by useEffect return null if component/page not mounted.

```
const [isMounted, setIsMounted] = useState(false);

  useEffect(() => {
    setIsMounted(true);
  }, []);

  if (!isMounted) return null;
```

[For more details official NextJs documentation](https://nextjs.org/docs/messages/react-hydration-error)
